agent/repowise.py

```
@@ -0,0 +1,283 @@
+
+        for name, keywords in pattern_indicators.items():
+            for kw in keywords:
+                cmd = f"grep -ri '{kw}' . --include='*.py' --include='*.js' --include='*.ts' --include='Dockerfile' --include='*.yaml' --exclude-dir={{.git,__pycache__,.venv,node_modules}} | head -n 1"
+                result = subprocess.run(cmd, shell=True, cwd=self.root, capture_output=True, text=True)
+                if result.stdout.strip():
+                    summary["patterns"].append(name)
+                    break
+
+        return summary
+
+    def get_context(self, targets: List[str], include: List[str] = ["source"]) -> str:
```

Ruff (0.15.12)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)